Platform/TigerlakeBoardPkg/BoardConfig.py

```diff
@@ -277,6 +277,7 @@ def GetPlatformDsc (self):
             'PchP2sbLib|Silicon/$(PCH_PKG_NAME)/Library/PchP2sbLib/PchP2sbLib.inf',
             'PchSbiAccessLib|Silicon/CommonSocPkg/Library/PchSbiAccessLib/PchSbiAccessLib.inf',
             'PlatformHookLib|Silicon/$(SILICON_PKG_NAME)/Library/PlatformHookLib/PlatformHookLib.inf',
+            'ResetSystemLib|Platform/$(BOARD_PKG_NAME)/Library/ResetSystemLib/ResetSystemLib.inf',
```

Comment:
I see that every other platform has this code, but why not use the BootloaderCommonPkg? 
 ResetSystemLib|BootloaderCommonPkg/Library/ResetSystemLib/ResetSystemLib.inf

Reply:
It is a good suggestion and I will rework the patch:
   1. update the common lib to implement for platform-specific reset.
   2. if a platform has a different implementation requirement for the "platform-specific reset", it will need to implement its own ResetSystemLib

Reply:
Summary: No rework will be made for the suggestion, but still thanks for the comment.

Detail:
I just re-examined the differences between common ResetSystemLib and TGL one. Unfortunately, TGL still needs to implement its own ResetSystemLib because of shutdown process.

And it may be not a good idea to add a "EfiResetPlatformSpecific" handler in this commit, because it will mix up 2 different changes in one commit. Let's only focus on TGL change for now. Thanks.

Reply:
It would be great to give info in the commit why the common reset lib could not meet the requirement and hance add this platform specific reset lib.

Reply:
> It would be great to give info in the commit why the common reset lib could not meet the requirement and hance add this platform specific reset lib.

Fixed with new commit message.